### PR TITLE
Initial diagrams addition.

### PR DESCRIPTION
**Pull request by ElijahCaine**
fixes issue #50 

## Changes in this PR.

[x] Adds example `source/figures/dataflow.dot` diagram file.
[x] Includes `dataflow.svg` in `draft/design.rst`.
[x] Adds `.buildfigures.sh` to the docs directory, builds `.dot` files
    and places them in `source/_static/`.
[x] Adds `figures` directory to the docs.
[x] Documented changes in `source/figures/README.md`.

## Testing this PR.

1. Build Docs.
2. Open Docs.
3. Navigate to `draft/design.html`.
4. Pretty graphs! Click it! Hover over nodes to get a tooltip!

@osuosl/devs

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -6,6 +6,7 @@ SPHINXOPTS    =
 SPHINXBUILD   = sphinx-build
 PAPER         =
 BUILDDIR      = build
+DOTCMD        = .buildfigures.sh
```
Comment (matthewrsj) [inline docs/Makefile]: Change this to `./.buildfigures.sh` for those of us that haven't added `.` to our path.

**Comment by matthewrsj** [comment]
I will +1 this once you address the very small issue above.

**Comment by alxngyn** [comment]
+1 docs will be docs

**Comment by Kennric** [comment]
+1

**Pull request merged**